dashboard/app/models/levels/level.rb

```diff
@@ -637,25 +637,38 @@ def clone_with_name(new_name, editor_experiment: nil)
   #   name_suffix if one exists.
   # @param [String] editor_experiment Optional value to set the
   #   editor_experiment property to on the newly-created level.
-  def clone_with_suffix(new_suffix, editor_experiment: nil)
+  # @param [Boolean] allow_existing Whether to return an existing level if one
+  #   with a matching name is found. If false, the suffix will be modified to
+  #   make the new name unique.
+  def clone_with_suffix(new_suffix, editor_experiment: nil, allow_existing: true)
```

Comment:
Would it be worth having a test verifying `allow_existing` being false works? I see a test for `next_unused_for_copy` and the level group cloning but not this method. A test for this might be redundant though so up to you